service/matching/matching_engine.go

```diff
@@ -536,7 +538,15 @@ pollLoop:
 				StartedEventId:             common.EmptyEventID,
 				Attempt:                    1,
 			}
-			return e.createPollWorkflowTaskQueueResponse(task, resp, opMetrics), nil
+
+			getHistResp, err := e.getHistoryForPollWorkflowTaskQueue(ctx, namespaceID, task, resp)
+			if err != nil {
+				// will notify query client that the query task failed
+				_ = e.deliverQueryResult(task.query.taskID, &queryResult{internalError: err})
```

Comment:
Shouldn't we log this error rather than swallowing it?

Reply:
It's not exactly swallowed because the error is still delivered to the query client.

I think the behavior should be the same as the error handling when we try to load mutable state on L521-L524.

Whether it would be useful to log the error in either case, I'm not sure. I would think any errors here are most likely to be transient and the client can just reissue another query, so maybe it is not necessary?

Reply:
I meant the error from deliverQueryResult itself